java-checks/src/main/java/org/sonar/java/filters/SpringFilter.java

```diff
@@ -0,0 +1,139 @@
+    "org.springframework.web.bind.annotation.DeleteMapping",
+    "org.springframework.web.bind.annotation.PatchMapping",
+    "org.springframework.context.annotation.Bean",
+    AUTOWIRED);
+
+  private static final List<String> S107_CLASS_ANNOTATION_EXCEPTIONS = Arrays.asList(
+    "org.springframework.stereotype.Component",
+    "org.springframework.context.annotation.Configuration",
+    "org.springframework.stereotype.Service",
+    "org.springframework.stereotype.Repository");
+
+  @Override
+  public Set<Class<? extends JavaCheck>> filteredRules() {
+    return Set.of(
+      /* S107_ */ TooManyParametersCheck.class,
+      /* S1185 */ MethodOnlyCallsSuperCheck.class,
+      /* S1258 */ AtLeastOneConstructorCheck.class,
+      /* S2226 */ ServletInstanceFieldCheck.class);
+  }
+
+  @Override
+  public void visitClass(ClassTree tree) {
+    excludeLinesIfTrue(isTransactional(tree), tree, MethodOnlyCallsSuperCheck.class);
+    excludeLinesIfTrue(hasAutowiredField(tree), tree.simpleName(), AtLeastOneConstructorCheck.class);
+    super.visitClass(tree);
+  }
+
+  @Override
+  public void visitVariable(VariableTree tree) {
+    excludeLinesIfTrue(isAutowired(tree), tree, ServletInstanceFieldCheck.class);
+    super.visitVariable(tree);
+  }
+
+  @Override
+  public void visitMethod(MethodTree tree) {
+    MethodSymbol symbol = tree.symbol();
+    if (tree.is(Tree.Kind.CONSTRUCTOR)) {
+      SymbolMetadata ownerMetadata = symbol.owner().metadata();
+      excludeLinesIfTrue(S107_CLASS_ANNOTATION_EXCEPTIONS.stream().anyMatch(ownerMetadata::isAnnotatedWith), tree.simpleName(), TooManyParametersCheck.class);
+    } else {
+      SymbolMetadata methodMetadata = symbol.metadata();
+      excludeLinesIfTrue(S107_METHOD_ANNOTATION_EXCEPTIONS.stream().anyMatch(methodMetadata::isAnnotatedWith), tree.simpleName(), TooManyParametersCheck.class);
+    }
+    super.visitMethod(tree);
+  }
+
+  private static boolean hasAutowiredField(ClassTree tree) {
+    return tree.members().stream()
+      .filter(member -> member.is(Tree.Kind.VARIABLE))
+      .map(VariableTree.class::cast)
+      .anyMatch(SpringFilter::isAutowired);
+  }
+
+  private static boolean isAutowired(VariableTree tree) {
+    return tree.symbol().metadata().isAnnotatedWith(AUTOWIRED)
+      // missing semantic
+      || hasUnknownAnnotationWithName(tree.modifiers(), "Autowired");
+  }
+
+  private static boolean isTransactional(ClassTree tree) {
+    return tree.symbol().metadata().isAnnotatedWith("org.springframework.transaction.annotation.Transactional")
+      // missing semantic
+      || hasUnknownAnnotationWithName(tree.modifiers(), "Transactional");
+  }
+
+  private static boolean hasUnknownAnnotationWithName(ModifiersTree modifiers, String annotation) {
+    // Token based check in case of missing semantic
+    List<AnnotationTree> annotations = modifiers.annotations();
+    if (annotations.isEmpty()) {
+      return false;
+    }
```

Comment:
```suggestion
```
Semantically we could remove this condition, and the final `anyMatch` would return false for empty lists of annotations, but I'm wondering if you put the condition to prevent the creation of `Stream` objects for every method with no annotations, so I don't have a strong opinion here

Reply:
Indeed, I wanted to avoid using streams, as I believe that a large majority of methods are not annotated. I'll add a comment.